core/app/[locale]/(default)/(faceted)/brand/[slug]/page.tsx

```diff
@@ -160,6 +163,39 @@ async function getPaginationInfo(props: Props): Promise<CursorPaginationInfo> {
   return pageInfoTransformer(brandSearch.products.pageInfo);
 }
 
+async function getShowCompare(props: Props) {
+  const { slug } = await props.params;
+
+  const variables = cachedBrandDataVariables(slug);
+  const data = await getBrandPageData(variables);
+
+  return data.settings?.storefront.catalog?.productComparisonsEnabled ?? false;
+}
+
+const cachedCompareProductIds = cache(async (props: Props) => {
+  const searchParams = await props.searchParams;
+
+  const compareIds =
+    typeof searchParams.compare === 'string' ? searchParams.compare.split(',') : [];
+
+  return { entityIds: compareIds.map((id: string) => Number(id)) };
```

Comment:
Should we use `nuqs` here?

Reply:
Do you mean create a `cache` using nuqs?

Reply:
Yup! And sharing the parsers where we set that data too

Reply:
So in this specific scenario, there is no need for cache, since the values are only set in a client component and we only read search params in the root Page file (no nested server components).

Per the docs:
```If you wish to access the searchParams in a deeply nested Server Component (ie: not in the Page component), you can use createSearchParamsCache to do so in a type-safe manner.```

However, the parsers may be worth sharing, looking into it.

Reply:
Updated to use parsers and loaders.